tests/integration/test_task_pool.py

```diff
@@ -2019,9 +2019,56 @@ async def test_remove_no_respawn(flow, scheduler, start, log_filter):
         # respawned as a result
         schd.pool.spawn_on_output(b1, TASK_OUTPUT_SUCCEEDED)
         assert log_filter(
-            log, contains='Not spawning 1/z - task removed'
+            log, contains='Not respawning 1/z - task was removed'
         )
         z1 = schd.pool.get_task(IntegerPoint("1"), "z")
         assert (
             z1 is None
         ), '1/z should have stayed removed (but has been added back into the pool'
+
+
+async def test_set_future_flow(flow, scheduler, start, log_filter):
+    """Manually-set outputs for new flow num must be recorded in the DB.
+
+    See https://github.com/cylc/cylc-flow/pull/6186
+
+    To trigger the bug, the flow must be new but the task must have been
+    spawned before in an earlier flow.
+
+    """
+    # Scenario: set c2:succeeded in a future flow so when b succeeds
+    # in that flow it will spawn c1 but not c2.
+    id_ = flow({
+        'scheduler': {
+            'allow implicit tasks': True
+        },
+        'scheduling': {
+            'cycling mode': 'integer',
+            'graph': {
+                'R1': 'a => b => c1 & c2 => z',
```

Comment:
IMO test becomes clearer if 

```suggestion
                'R1': 'a => good & failonce => z',
```

And is the task `b` necessary?

Reply:
> IMO test becomes clearer if

Well that makes it look as if success and failure are important for the test, which they're not. What matters is that only one of them gets respawned in the new flow, after the other one get manually set to succeeded (in the new flow).

> And is the task b necessary?

Actually neither b nor z are needed. They were just to make the original functional case easier to manage. I'll remove them.